lib/author.rb

```diff
@@ -0,0 +1,34 @@
+class Author
+  attr_reader :name,
+              :books
+
+  def initialize(info)
+    @info = info
+    @name = info[:first_name] + " " + info[:last_name]
+    @books = []
+  end
+
+  def write(title, publication_date)
```

Comment:
The way that you have this method written isn't quite doing what you expect (I think).  The way it is now, the first book that an author writes will be create just fine, but the second, third and so on will be created without an author.  In your `else` branch, you are setting up the `:title` and `:publication_date` keys, but not the `:author_first_name` and `:author_last_name`.  At line 24, if you pry in, your `info` hash will look like this:

```ruby
info = {
  first_name: 'Charlotte',
  last_name: 'Bronte',
  title: 'Jane Eyre',
  publication_date: 'October 16, 1847'
}
```

Based on the way your Book is defined, the above info hash will create a book with a title and publication year, but no author.

A simpler solution might look something like this:

```ruby
def write(title, publication_date)
  book = Book.new({
    author_first_name: @first_name,
    author_last_name: @last_name,
    title: title,
    publication_date: publication_date
  })
  @books << book
  book
end
```